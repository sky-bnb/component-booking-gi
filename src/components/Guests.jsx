import React from 'react';
import styled from 'styled-components';
import GuestsButton from './GuestsButton.jsx';
import GuestsDropdown from './GuestsDropdown.jsx';

export const StyledGuests = styled.div`
  margin-bottom: 16px;
  position: relative;
`;

const StyledGuestsLabel = styled.label`
  margin: 0px;
  word-wrap: break-word;
  font-family: 'Quicksand', sans-serif;
  font-size: 12px;
  font-weight: 600;
  line-height: 1.3333333333333333em;
  color: #484848;
`;

const Guests = (props) => {
  const {
    isOpen, toggle, numAdults, numChildren, numInfants, guestMax, maxInfants,
    updateNumAdults, updateNumChildren, updateNumInfants,
  } = props;

  return (
    <StyledGuests>
      <StyledGuestsLabel>Guests</StyledGuestsLabel>
      <GuestsButton
        isOpen={isOpen}
        toggle={toggle}
        numAdults={numAdults}
        numChildren={numChildren}
        numInfants={numInfants}
      />
      {isOpen ? (
        <GuestsDropdown
          toggle={toggle}
          numAdults={numAdults}
          numChildren={numChildren}
          numInfants={numInfants}
          guestMax={guestMax}
          maxInfants={maxInfants}
          updateNumAdults={updateNumAdults}
          updateNumChildren={updateNumChildren}
          updateNumInfants={updateNumInfants}
        />
      ) : null}
    </StyledGuests>
  );
};

export default Guests;
