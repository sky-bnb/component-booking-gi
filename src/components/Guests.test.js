import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Guests, { StyledGuests } from './Guests.jsx';

describe('Guests Component', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<Guests />));

  it('renders the Guests component', () => {
    expect(wrapper.find(StyledGuests).length).toBe(1);
  });

  it('matches the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
