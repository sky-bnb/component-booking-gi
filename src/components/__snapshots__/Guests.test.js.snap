// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guests Component matches the snapshot 1`] = `
<Guests__StyledGuests>
  <Guests__StyledGuestsLabel>
    Guests
  </Guests__StyledGuestsLabel>
  <GuestsButton />
</Guests__StyledGuests>
`;
